packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -565,6 +565,35 @@ the configured path by running this command: flutter config --android-studio-dir
     }
   }
 
+  static Version? _parseVersion(String? text) {
+    Match? match;
+    // Example string fort EAP: EAP AI-242.21829.142.2422.12358220
+    // We try to capture 2422 here.
+    final RegExp intellijPattern = RegExp(r'EAP\s+[A-Z]{2}-\d+\.\d+\.\d+\.(\d+)\.\d+');
+    match = intellijPattern.firstMatch(text ?? '');
+
+    if (match == null) {
+      return Version.parse(text);
+    }
+
+    final String? matched = match.group(1);
+
+    // length of 4 is because that how version is encrypted: first two digits
+    // for year (part of major version), third for minor version, fourth for patch.
+    if (matched == null || matched.length != 4) {
+      return null;
+    }
+
+    final int? major = int.tryParse('20${matched[0]}${matched[1]}');
+    final int? minor = int.tryParse(matched[2]);
+    final int? patch = int.tryParse(matched[3]);
+    if (major == null || minor == null || patch == null) {
+      return null;
+    } else {
```

Comment:
The else is redundant here

```suggestion
if (major == null || minor == null || patch == null) {
  return null;
}

return Version(major, minor, patch);
```

Reply:
Certainly! Thanks